docs/contribute.md

```diff
@@ -0,0 +1,340 @@
+# Contribute to HashiCorp documentation
```

Comment:
Consul and Nomad have existing Website contribution guides that could be adapted for this page.

Here's a link to Consul's as an example: https://github.com/hashicorp/consul/blob/main/website/README.md

I'd also strongly suggest a Table of Contents at the start, like Consul's example, to make it easier to navigate the guide.